src/model/EnergyManagementSystemActuator.cpp

```diff
@@ -108,51 +101,78 @@ namespace model {
     }
 
     bool EnergyManagementSystemActuator_Impl::setActuatedComponent(const ModelObject& modelObject) {
-      return setPointer(OS_EnergyManagementSystem_ActuatorFields::ActuatedComponentName, modelObject.handle());
-    }
-
-    bool EnergyManagementSystemActuator_Impl::setThermalZone(const ThermalZone& thermalZone) {
-      return setPointer(OS_EnergyManagementSystem_ActuatorFields::ZoneName, thermalZone.handle());
-    }
-
-    bool EnergyManagementSystemActuator_Impl::setSpace(const Space& space) {
-      //boost::optional<ThermalZone> tz;
-      auto tz = space.thermalZone();
-      if (tz) {
-        return setPointer(OS_EnergyManagementSystem_ActuatorFields::ZoneName, tz.get().handle());
-      } else {
-        LOG(Warn, "Warning, Space object" << space.briefDescription() << " does not have a ThermalZone object.")
-        return false;
+      if (!modelObject.optionalCast<SpaceLoadInstance>()) {
+        resetZoneOrSpace();
```

Comment:
in setActuatedComponent, if the new comp is not a SpaceLoadInstance, remove the Zone or Space